lib/double_doc/task.rb

```diff
@@ -57,48 +55,17 @@ def initialize(task_name, options)
         directory(dst.to_s)
       end
 
+      roots = Array(options[:root])
+      roots << Rake.original_dir if roots.empty?
+
       desc "Generate markdown #{html_dst ? 'and HTML ' : ''}DoubleDoc documentation"
       generated_task = task(task_name => destinations) do |t, args|
-        roots = Array(options[:root])
-        roots << Rake.original_dir if roots.empty?
-        roots << options.fetch(:import, {})
-
-        import_handler = DoubleDoc::ImportHandler.new(*roots)
-
-        sources = md_srcs.map do |source|
-          if source =~ /\*/
-            import_handler.load_paths.map do |path|
-              Dir.glob(File.join(path, source))
-            end
-          else
-            import_handler.find_file(source).path
-          end
-        end.flatten.uniq
-
-        generated_md_files = []
-
-        sources.each do |src|
-          dst = md_dst + File.basename(src)
-          puts "#{src} -> #{dst}"
-
-          if src.to_s =~ /\.md$/
-            body = import_handler.resolve_imports(File.new(src))
-          else
-            body = File.read(src)
-          end
-
-          File.open(dst, 'w') do |out|
-            out.write(body)
-          end
-
-          generated_md_files << dst
-        end
-
-        if html_dst || args[:html_destination]
-          html_generator = DoubleDoc::HtmlGenerator.new(generated_md_files, options.merge(args))
-          html_generator.generate
-        end
+        client = DoubleDoc::Client.new(options[:sources], options.merge({
```

Comment:
I didn't merge because i thought someone might have specified `:html_destination` in the options and didn't want to blow it away. From what I could tell it's only using `args` for generating [`github_pages`](https://github.com/zendesk/double_doc/blob/doubledoc-client/lib/double_doc/task.rb#L83). If it's not a problem blowing away options, merging is fine with me.